internal/pkg/gateway/commit/commitadapter.go

```diff
@@ -28,13 +27,32 @@ func (adapter *PeerAdapter) CommitNotifications(done <-chan struct{}, channelNam
 	return channel.Ledger().CommitNotificationsChannel(done)
 }
 
-func (adapter *PeerAdapter) TransactionStatus(channelName string, transactionID string) (peerproto.TxValidationCode, error) {
+func (adapter *PeerAdapter) TransactionStatus(channelName string, transactionID string) (*Status, error) {
 	channel, err := adapter.channel(channelName)
 	if err != nil {
-		return 0, err
+		return nil, err
+	}
+
+	ledger := channel.Ledger()
+
+	status, err := ledger.GetTxValidationCodeByTxID(transactionID)
+	if err != nil {
+		return nil, err
 	}
 
-	return channel.Ledger().GetTxValidationCodeByTxID(transactionID)
+	block, err := ledger.GetBlockByTxID(transactionID)
+	if err != nil {
+		return nil, err
+	}
+	if block.Header == nil {
+		return nil, errors.Errorf("missing header for block containing transaction %s", transactionID)
+	}
```

Comment:
This appears a bit weird to me, as you check whenever the header is not nil, while not checking whenever a block is not nil. not something expected to happen here. However, since you are already adding such a check for the header, I would expect to have a similar check for the block itself.

Reply:
Won't `GetBlockByTxID()` return a non-nil error if there is no block?

Reply:
Well, it will also return a valid non-nil header in case there are no errors. I guess it depends on what was your intent while adding this check for the non-nil header.

See: 

```go
func deserializeBlock(serializedBlockBytes []byte) (*common.Block, error) {
	block := &common.Block{}
	var err error
	b := newBuffer(serializedBlockBytes)
	if block.Header, err = extractHeader(b); err != nil {
		return nil, err
	}
	if block.Data, _, err = extractData(b); err != nil {
		return nil, err
	}
	if block.Metadata, err = extractMetadata(b); err != nil {
		return nil, err
	}
	return block, nil
}
```

How the block is actually being fetched. Now, this is a nit comment, not a real show stopper, just for the sake of consistency and clarity. Like, why did you check for non-nil header?

Reply:
It's there because any time I deliver code that deals with protobufs and doesn't explicitly check for nil, it gets flagged in review comments. I'm quite happy to remove the check if that's the preferred approach.

Reply:
I've pushed an update that removes the nil check on the block header

Reply:
My initial comment and intent were not to remove the check, but for sanity and clearness, add the check to ensure the block is not nil neither. ;)

Reply:
I'm not convinced we can reasonably expect `GetBlockByTxID()` to return `nil, nil`. The error response is there to indicate that a block isn't returned. It seems confusing and non-idiomatic to check both the value and the associated error response for nil.